element-connector/src/main/java/org/eclipse/californium/elements/TcpCorrelationContext.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - add support for correlation context to provide
+ *                                      additional information to application layer for
+ *                                      matching messages (fix GitHub issue #1)
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+/**
+ * A correlation context that explicitly supports TCP specific properties.
+ */
+public class TcpCorrelationContext extends MapBasedCorrelationContext {
+
+	public static final String KEY_CONNECTION_ID = "CONNECTION_ID";
```

Comment:
> So, why do you want to propagate this Netty property up the layers?

As I understood the intention of the correlation context, its mainly used to ensure, that a response is received with the same context as the request was sent.

So, though a TCP connection may be reopened, I think there is a requirement to ensure, that a response is received in the same and not in the reopend connection. The idea is, that netty overs that information. By now, its not tested.